### PR TITLE
fix: [IDP-1305]: Audit trials - host proxy

### PR DESCRIPTION
**Pull request by devesh-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54043)

### CODE REVIEW

**Review by vikyathharekal — COMMENTED** [review]
> Reviewed 10 of 13 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54043)* status: 13 file(s) to review. 0 of 2 approvals obtained. 

___
*[`idp-service/src/main/java/io/harness/idp/audittrials/eventhandlers/ProxyHostDetailsEventHandler.java` line 38 at r2](https://reviewable.io/reviews/harness/harness-core/54043#-NgXQ7TN8N5voCJ7OOuG:-NgXQ7TN8N5voCJ7OOuH:b-docpge) ([raw file](https://github.com/harness/harness-core/blob/78425e7f639c589a7658462e5e742f363b9c85c2/idp-service/src/main/java/io/harness/idp/audittrials/eventhandlers/ProxyHostDetailsEventHandler.java#L38)):*
> ```Java
> 
> @Slf4j
> public class ProxyHostDetailsEventHandler implements OutboxEventHandler {
> ```

nit: IDP team
___
*[`idp-service/src/main/java/io/harness/idp/audittrials/eventhandlers/dtos/ProxyHostDTO.java` line 7 at r2](https://reviewable.io/reviews/harness/harness-core/54043#-NgXQoHX2_0PJaKui0_5:-NgXQoHX2_0PJaKui0_6:b-i4mb8a) ([raw file](https://github.com/harness/harness-core/blob/78425e7f639c589a7658462e5e742f363b9c85c2/idp-service/src/main/java/io/harness/idp/audittrials/eventhandlers/dtos/ProxyHostDTO.java#L7)):*
> ```Java
>  * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
>  */
> package io.harness.idp.audittrials.eventhandlers.dtos;
> ```

we might have to rename this to audittrails



_Code quote:_
```Java
audittrials
```
___
*[`idp-service/src/main/java/io/harness/idp/configmanager/service/PluginsProxyInfoServiceImpl.java` line 119 at r2](https://reviewable.io/reviews/harness/harness-core/54043#-NgXWVev1P9o0ATXtq9f:-NgXWVev1P9o0ATXtq9g:b-lgwlyl) ([raw file](https://github.com/harness/harness-core/blob/78425e7f639c589a7658462e5e742f363b9c85c2/idp-service/src/main/java/io/harness/idp/configmanager/service/PluginsProxyInfoServiceImpl.java#L119)):*
> ```Java
> 
>     // removing the newly added proxy host from the list for the update case
>     newProxyHostDetails.removeAll(newlyAddedProxyHotsDetails);
> ```

How about naming this as proxyHostDetailsToUpdate or something similar?
This looks like we are removing new from new 



_Code quote:_
```Java
newProxyHostDetails
```
___
*[`idp-service/src/main/java/io/harness/idp/configmanager/service/PluginsProxyInfoServiceImpl.java` line 278 at r2](https://reviewable.io/reviews/harness/harness-core/54043#-NgXVat1ByLXHBTBfLJs:-NgXVat1ByLXHBTBfLJt:b84iqu) ([raw file](https://github.com/harness/harness-core/blob/78425e7f639c589a7658462e5e742f363b9c85c2/idp-service/src/main/java/io/harness/idp/configmanager/service/PluginsProxyInfoServiceImpl.java#L278)):*
> ```Java
>         returnList.add(proxyHostDetail);
>       }
>       pluginsProxyInfoRepository.saveAll(pluginsProxyInfoEntities);
> ```

As discussed, we need to return IDs after DB save



_Code quote:_
```Java
saveAl
```

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 11 of 11 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54043)* status: 20 file(s) to review. 1 of 2 approvals obtained.

**Comment by devesh-harness** [comment]
trigger uts

**Comment by devesh-harness** [comment]
trigger ut5, ut6, ut7, ut8, ut9

**Comment by devesh-harness** [comment]
trigger ut9

**Review by vikyathharekal — APPROVED** [review]

File: idp-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -2227,45 +2227,53 @@ components:
...
           items:
             x-stoplight:
-              id: nallsx0kcechh
+              id: zvnhr2daor8lr
             type: string
+        identifier:
+          type: string
+          x-stoplight:
+            id: soyi91ysmnt8u
+        pluginId:
```
Comment (mankrit-singh) [inline idp-service/contracts/openapi/v1/openapi.yaml:2254]: please fix the casing - plugin_id

**Review by mankrit-singh — APPROVED** [review]

File: idp-service/contracts/openapi/v1/openapi.yaml (hunk 1 above)
Comment (mankrit-singh) [reply idp-service/contracts/openapi/v1/openapi.yaml:2254]: please take it up in the next PR

**Comment by devesh-harness** [comment]
trigger review

**Pull request merged**